src/routing/core/messaging/handling/mod.rs

```diff
@@ -53,7 +53,11 @@ impl Core {
 
         // Check if the message is for us.
         let in_dst_location = msg.dst().contains(&self.node.name(), self.section.prefix());
-        if !in_dst_location || msg.dst().is_section() {
```

Comment:
removing `msg.dst().is_section()` means when the sender knows only part of the receivers, the msg won't reach all.
For node->section msgs, it may cause issue.
Meanwhile, I agree this will reduce certain amount of traffic. 
Worth to at least put a comment here, mark as `TODO` or `FIXME`